ios/Capacitor/Capacitor/WebViewAssetHandler.swift

```diff
@@ -80,7 +81,7 @@ open class WebViewAssetHandler: NSObject, WKURLSchemeHandler {
                 headers["Content-Range"] = "bytes \(fromRange)-\(toRange)/\(totalSize)"
                 headers["Content-Length"] = String(data.count)
                 let response = HTTPURLResponse(url: localUrl, statusCode: 206, httpVersion: nil, headerFields: headers)
-                urlSchemeTask.didReceive(response!)
+                lock(urlSchemeTask, { urlSchemeTask.didReceive(response!) })
```

Comment:
Mentioned in the other comment, this isn't actually locking any access to anything. The best way to achieve what you are wanting is to do this:
```
class ConcurrentTasks {
  func withTask(_ task: WKURLSchemeTask, action: () -> Void) {
    if !tasks.contains(task.hash) {
      action()
    }
  }
}
```
This will block anything trying to write to the `stoppedTasks` until it is finished.